### PR TITLE
Add citation details for new paper

### PR DESCRIPTION
**Pull request by Andrew-S-Rosen**
This PR adds citation details for the new JCP paper. I added it to the docs as well as a badge to the README. In principle, one could also add a `citation.cff` file, but I didn't want to bother with that.

Note that I have this PR set to merge into `main`. Let me know if you'd like that to be `master` instead.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,8 @@ See https://github.com/cclib/cclib/issues/1395 for more information.
 - We do not expect to make any further tagged or versioned releases on the `master` branch.
 - This message will disappear when the final release of 2.0, after any alphas/release candidates/etc. is made.
 
-[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.8280878.svg)](https://doi.org/10.5281/zenodo.8280878)
+[![Paper](https://img.shields.io/badge/Paper-10.1063/5.0216778-blue)](https://doi.org/10.1063/5.0216778)
```
Comment (langner) [inline README.md:11]: Perhaps also put the old paper here for the records.

Comment (Andrew-S-Rosen) [reply README.md:11]: I left the old paper in the docs. Would you like it in the badges too?

Comment (berquist) [reply README.md:11]: I'm ok with it only in the docs.

File: doc/sphinx/index.rst [hunk 2]
```diff
@@ -49,9 +49,14 @@ About cclib
...
 
-If you use cclib in your scientific work, please support our work by adding a reference to the following article:
+If you use cclib (version 2.0+) in your scientific work, please support our work by adding a reference to the following article:
 
-|           N\. M\. O'Boyle, A\. L\. Tenderholt, K\. M\. Langner, *cclib: a library for package-independent computational chemistry algorithms*, J. Comp. Chem. 29 (5), pp. 839-845, **2008** (DOI_).
+|           E\. Berquist, A\.Dumi, S\. Upadhyay, O\.D\. Abarbanel, M\. Cho, S\. Gaur, V\. Hugo Cano Gil, G\.R\. Hutchinson, O\.S\. Lee, A\.S\. Rosen, S\. Schamnad, F\.S\.S Schneider, C\. Steinmann, M\. Stolyarchuk, J\.E\. Vandezande, W\. Zak, K\.M\. Langner, *cclib 2.0: An updated architecture for interoperable computational chemistry*, J. Chem. Phys. 161, 042501, **2024** (DOI_).
```
Comment (berquist) [inline doc/sphinx/index.rst:54]:
```suggestion
|           E\. Berquist, A\. Dumi, S\. Upadhyay, O\. D\. Abarbanel, M\. Cho, S\. Gaur, V\. Hugo Cano Gil, G\. R\. Hutchinson, O\. S\. Lee, A\. S\. Rosen, S\. Schamnad, F\. S\. S\. Schneider, C\. Steinmann, M\. Stolyarchuk, J\. E\. Vandezande, W\. Zak, K\. M\. Langner, *cclib 2.0: An updated architecture for interoperable computational chemistry*, J. Chem. Phys. 161, 042501, **2024** (DOI_).
```

**Review by berquist — CHANGES_REQUESTED** [review]
I'll approve once I look at the build output.

`main` is good.

I'll add `CITATION.cff` later.

**Review by berquist — APPROVED** [review]

**Pull request merged**